modules/dashboard/ajax/get_scan_line_data.php

```diff
@@ -25,94 +25,102 @@
 
 $DB = Database::singleton();
 
-$scanData           = array();
-$scanStartDate      = $DB->pselectOne(
+$scanData      = array();
+$scanStartDate = $DB->pselectOne(
     "SELECT MIN(pf.Value) 
      FROM parameter_file pf 
      JOIN parameter_type pt USING (ParameterTypeID) 
      WHERE pt.Name='acquisition_date'",
     array()
 );
-$scanEndDate        = $DB->pselectOne(
+$scanEndDate   = $DB->pselectOne(
     "SELECT MAX(pf.Value) 
      FROM parameter_file pf 
      JOIN parameter_type pt USING (ParameterTypeID) 
      WHERE pt.Name='acquisition_date'",
     array()
 );
-$scanData['labels']
-    = createLineChartLabels($scanStartDate, $scanEndDate);
-$list_of_sites      = Utility::getSiteList(true, false);
+
+// Run a query to get all the data. Order matters to ensure that the labels
+// are calculated in the correct order.
+$data = $DB->pselect(
+    "SELECT s.CenterID, 
+        CONCAT(MONTH(pf.Value), '-', YEAR(pf.Value)) as datelabel,
+        COUNT(distinct s.ID) as count
+    FROM files f
+        LEFT JOIN parameter_file pf USING (FileID)
+        LEFT JOIN session s ON (s.ID=f.SessionID)
+        JOIN parameter_type pt USING (ParameterTypeID)
+    WHERE pt.Name='acquisition_date'
+    GROUP BY MONTH(pf.Value), YEAR(pf.Value), s.CenterID
+    ORDER BY YEAR(pf.Value), MONTH(pf.Value), s.CenterID",
+    array()
+);
+
+$scanData['labels'] = createLineChartLabels($data);
+
+$list_of_sites = Utility::getSiteList(true, false);
 foreach ($list_of_sites as $siteID => $siteName) {
     $scanData['datasets'][] = array(
         "name" => $siteName,
-        "data" => getScanData($siteID, $scanData['labels']),
+        "data" => getScanData($data, $siteID, $scanData['labels'])
     );
 }
 
-print json_encode($scanData);
-
-return 0;
-
 /**
- * Create chart labels (dates)
+ * Massages the data passed into the format expected by C3.
```

Comment:
What is the meaning of `C3` ?

Reply:
It's the graphing library used by the dashboard

Reply:
Ok thanks, Would it be possible to add `... by the C3 library` in that case?
I will approve now as it is really minor.

Reply:
okay, changed.